application-admintools-default/src/test/java/com/xwiki/admintools/internal/data/identifiers/CurrentServerTest.java

```diff
@@ -0,0 +1,162 @@
+import org.xwiki.test.junit5.mockito.MockComponent;
+
+import com.xwiki.admintools.ServerIdentifier;
+import com.xwiki.admintools.configuration.AdminToolsConfiguration;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit test for {@link CurrentServer}
+ *
+ * @version $Id$
+ */
+@ComponentTest
+public class CurrentServerTest
+{
+    @InjectMockComponents
+    private CurrentServer currentServer;
+
+    @MockComponent
+    private Provider<List<ServerIdentifier>> supportedServers;
+
+    @MockComponent
+    private ServerIdentifier mockServerIdentifier;
+
+    @MockComponent
+    @Named("default")
+    private AdminToolsConfiguration adminToolsConfig;
+
+    @BeforeComponent
+    void setUp(){
+        // Mock the list of supported servers
+        List<ServerIdentifier> mockServerIdentifiers = new ArrayList<>();
+        mockServerIdentifiers.add(mockServerIdentifier);
+        when(supportedServers.get()).thenReturn(mockServerIdentifiers);
+        when(mockServerIdentifier.isUsed()).thenReturn(true);
+    }
+
+    @Test
+    void Initialize() throws InitializationException
+    {
+        // Mock the behavior of adminToolsConfig
+        when(adminToolsConfig.getServerPath()).thenReturn("exampleServerPath");
+
+        // Call the initialize method
+        currentServer.initialize();
+
+        // Verify that the currentServerIdentifier is set correctly
+        assertEquals(mockServerIdentifier, currentServer.getCurrentServer());
+    }
+
+    @Test
+    void updateCurrentServerAfterInitializationNotFound() throws InitializationException
+    {
+        // Mock the behavior of adminToolsConfig
+        when(adminToolsConfig.getServerPath()).thenReturn("exampleServerPath");
+
+        // Create a mock implementation of ServerIdentifier
+        ServerIdentifier mockServerIdentifier = mock(ServerIdentifier.class);
+
+        // Mock the list of supported servers
+        List<ServerIdentifier> mockServerIdentifiers = new ArrayList<>();
+        mockServerIdentifiers.add(mockServerIdentifier);
+        when(supportedServers.get()).thenReturn(mockServerIdentifiers);
+        when(mockServerIdentifier.isUsed()).thenReturn(false);
+
+        // Call the initialize method
+        currentServer.initialize();
+
+        // Verify that the currentServerIdentifier was not found
+        assertNull(currentServer.getCurrentServer());
+
+        // Mock the behaviour of serverIdentifier
+        when(mockServerIdentifier.isUsed()).thenReturn(true);
+
+        // Call the updateCurrentServer method
+        currentServer.updateCurrentServer();
+
+        // Verify that the currentServerIdentifier is set correctly
+        assertEquals(mockServerIdentifier, currentServer.getCurrentServer());
+    }
+
+    @Test
+    void initializeNotFound() throws InitializationException
```

Comment:
I think this test is the same with the `initializeWithServerNotUsed` I mentioned above, since here you just test the case when there is a serverIdentifier, only that `isUsed=false` (you didn't added this setup explicitly, but you should have since that method could return either true or false)